gap/semigroups/semitrans.gi

```diff
@@ -84,24 +84,37 @@ InstallMethod(DirectProductOp,
 "for a list and a transformation monoid as semigroup",
 [IsList, IsTransformationSemigroup], 1, # to beat the next method
 function(list, S)
-  local gens, deg, m, i, x;
+  local gens, deg, one, m, next, i, x;
 
   # Check the arguments.
   if IsEmpty(list) then
     ErrorNoReturn("Semigroups: DirectProductOp: usage,\n",
                   "the first argument must be a non-empty list,");
   elif not ForAll(list, T -> IsTransformationSemigroup(T)
-                  and IsMonoidAsSemigroup(T)) then
+                             and IsMonoidAsSemigroup(T)) then
     TryNextMethod();
   fi;
 
   gens := ShallowCopy(GeneratorsOfSemigroup(list[1]));
```

Comment:
These first generators also need to be determined in terms of `one`.  At the moment, this assumes that all semigroups after `list[i]` have `IdentityTransformation` as their identity.